cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AppFabricServiceManager.java

```diff
@@ -49,14 +50,18 @@
   private static final long ELECTION_PARTICIPANTS_TIMEOUT_MS = 2000L;
 
   private final InetAddress hostname;
-  private LeaderElectionInfoService electionInfoService;
   private final Map<String, LogEntry.Level> oldLogLevels;
+  private final AppFabricServer appFabricServer;
+
+  private LeaderElectionInfoService electionInfoService;
 
   @Inject
-  public AppFabricServiceManager(@Named(Constants.Service.MASTER_SERVICES_BIND_ADDRESS) InetAddress hostname) {
+  public AppFabricServiceManager(@Named(Constants.Service.MASTER_SERVICES_BIND_ADDRESS) InetAddress hostname,
+                                 AppFabricServer appFabricServer) {
```

Comment:
Make sure this is bound as singleton

Reply:
I added the AppFabricServer to be @Singleton at here: https://github.com/cdapio/cdap/pull/11086/files#diff-024664c5526c921c18a5f120e7e4edd1R64